src/web/components/Nav/StickNavTest/ExpandedMenu/ExpandedMenu.tsx

```diff
@@ -0,0 +1,148 @@
+import { textSans } from '@guardian/src-foundations/typography';
+import { from, until } from '@guardian/src-foundations/mq';
+
+import { Display } from '@guardian/types';
+import {
+	buildID,
+	navInputCheckboxId,
+} from '@root/src/web/components/Nav/config';
+import { extractNAV } from '@root/src/model/extract-nav';
+
+import { getZIndex } from '@root/src/web/lib/getZIndex';
+import { Columns } from './Columns';
+import { ShowMoreMenu } from './ShowMoreMenu';
+import { VeggieBurgerMenu } from './VeggieBurgerMenu';
+
+const mainMenuStyles = (ID: string) => css`
+	background-color: ${brandBackground.primary};
+	box-sizing: border-box;
+	${textSans.large()};
+	left: 0;
+	margin-right: 29px;
+	padding-bottom: 24px;
+	top: 0;
+	${getZIndex('stickyNav')}
+	overflow: hidden;
+
+	/*
+        IMPORTANT NOTE:
+        we need to specify the adjacent path to the a (current) tag
+        to apply styles to the nested tabs due to the fact we use ~
+        to support NoJS
+    */
+	/* stylelint-disable-next-line selector-type-no-unknown */
+	${`#${buildID(ID, navInputCheckboxId)}`}:checked ~ div & {
+		${from.desktop} {
+			display: block;
+			overflow: visible;
+		}
+	}
+
+	${from.desktop} {
+		display: none;
+		position: absolute;
+		padding-bottom: 0;
+		padding-top: 0;
+		top: 100%;
+		left: 0;
+		right: 0;
+		width: 100%;
+		@supports (width: 100vw) {
+			left: 50%;
+			right: 50%;
+			width: 100vw;
+			margin-left: -50vw;
+			margin-right: -50vw;
+		}
+	}
+
+	/* refer to comment above */
+	/* stylelint-disable */
+	${`#${buildID(ID, navInputCheckboxId)}`}:checked ~ div & {
+		${until.desktop} {
+			transform: translateX(
+				0%
+			); /* when translateX is set to 0% it reapears on the screen */
+		}
+	}
+
+	${until.desktop} {
+		transform: translateX(
+			-110%
+		); /* the negative translateX makes the nav hide to the side */
+		transition: transform 0.4s cubic-bezier(0.23, 1, 0.32, 1);
+		box-shadow: 3px 0 16px rgba(0, 0, 0, 0.4);
+		bottom: 0;
+		height: 100%;
+		overflow: auto;
+		padding-top: 6px;
+		position: fixed;
+		right: 0;
+		will-change: transform;
+	}
+
+	${from.mobileMedium} {
+		margin-right: 29px;
+	}
+	${from.mobileLandscape} {
+		margin-right: 70px;
+	}
+	${from.tablet} {
+		margin-right: 100px;
+	}
+`;
+
+const fetchNavData = (
+	ajaxUrl: string,
+	edition: Edition,
+): Promise<SimpleNavType> => {
+	const url = `${ajaxUrl}/nav/${edition.toLowerCase()}.json`;
+	return fetch(url)
+		.then((resp) => resp.json())
+		.then((json) => extractNAV(json));
+};
+
+const enrich = (currentNavLinkTitle: string, nav: SimpleNavType) => {
+	return extractNAV({ currentNavLinkTitle, ...nav });
+};
+
+export const ExpandedMenu: React.FC<{
+	display: Display;
+	ID: string;
+	currentNavLinkTitle: string;
+	expand: boolean;
+	edition: Edition;
+	ajaxUrl: string;
+}> = ({ display, currentNavLinkTitle, ID, expand, edition, ajaxUrl }) => {
+	const [navData, setNavData] = useState<NavType | undefined>(undefined);
+
+	useEffect(() => {
```

Comment:
We have some existing patterns for fetching data into react components that offer a lot of benefit in terms of features and might be easier here? They're in `api.ts`, you can see examples with `useApi`

Reply:
In fact, this is a bit more difficult. The API call is made iff the nav is expanded. I think we could refactor the `useApi` helper to account for this (perhaps with a `when: boolean`) parameter, but will leave this until/if the test is successful.

Reply:
> leave this until/if the test is successful

👍 I think if we're reviewing this code after the test this is one area we can come back to

Thinking towards this, one solution might be to use composition. If you had the `ExpandedMenu` hidden behind a state flag like `isExpanded` inside some parent, then `ExpandedMenu` would only be rendered when that flag were true and then the useEffect and the subsequent api call would only get triggered at that point.

Reply:
Ah yes that is a nice way to do it.